posts/bayesian-model-comparison/index.qmd

```diff
@@ -225,32 +212,53 @@ RSKF = RepeatedStratifiedKFold(
   n_splits=n_folds, n_repeats=n_repetitions, random_state=23
 )
 
-models = {"svm": SVC, "rf": RandomForestClassifier}
+model_1 = "RF"
+model_2 = "MLP"
+models = {
+  model_1: RandomForestClassifier(),
+  model_2: MLPClassifier(alpha=1, max_iter=1_000)
+}
 
 model_scores = {}
+model_times = {}
 
 for model_name, model in models.items():
+  times_fit = []
+  times_predict = []
   scores = []
-  for i, (train_indices, test_indices) in enumerate(RSKF.split(X_cancer, Y_cancer)):
-    x_train, x_test = X_cancer[train_indices], X_cancer[test_indices]
-    y_train, y_test = Y_cancer[train_indices], Y_cancer[test_indices]
+  for i, (train_indices, test_indices) in enumerate(RSKF.split(X, y)):
+    X_train, X_test = X[train_indices], X[test_indices]
+    y_train, y_test = y[train_indices], y[test_indices]
     
-    clf = model().fit(x_train, y_train)
-    y_pred = clf.predict(x_test)
+    time_start = perf_counter()
+    clf = model.fit(X_train, y_train)
+    time_end = perf_counter()
+    time_fit = time_end - time_start
+
+    time_start = perf_counter()
+    y_pred = clf.predict(X_test)
+    time_end = perf_counter()
+    time_predict = time_end - time_start
 
     f1 = f1_score(y_test, y_pred)
 
+    times_fit.append(time_fit)
+    times_predict.append(time_predict)
     scores.append(f1)
   
   model_scores[model_name] = np.asarray(scores)
+  model_times[model_name] = {
+    "fit": times_fit,
+    "predict": times_predict
+  }
 ```
 
-Now that we have the scores for each model let's perform the test. 
-We'll first need to compute the array of differences, $\delta$ and then we're in business.
+Now that we have the scores for each model, let's perform the test. 
+We'll first need to compute the array of differences, $\delta$, and then we're in business.
 
 ```{python}
-# * create array of differences; we do rf - svm
-delta = model_scores["rf"] - model_scores["svm"]
+# * create array of differences; we do model_2 (MLP) - model_1 (RF)
+delta = model_scores[model_2] - model_scores[model_1]
```

Comment:
Should we print the mean, var and DoF below, when we compute these? Atm they aren't shown and maybe it would help?